pwndbg/heap/ptmalloc.py

```diff
@@ -563,14 +563,8 @@ def largebins(self, arena_addr=None):
 
 
     def is_initialized(self):
-        """
-        malloc state is initialized when a new arena is created. 
-            https://sourceware.org/git/?p=glibc.git;a=blob;f=malloc/malloc.c;h=96149549758dd424f5c08bed3b7ed1259d5d5664;hb=HEAD#l1807
-        By default main_arena is partially initialized, and during the first usage of a glibc allocator function some other field are populated.
-        global_max_fast is one of them thus the call of set_max_fast() when initializing the main_arena, 
-        making it one of the ways to check if the allocator is initialized or not.
-        """
-        return self.global_max_fast != 0
+        addr = pwndbg.symbol.address('__libc_malloc_initialized')
+        return pwndbg.memory.u32(addr) == 1
```

Comment:
> 	* malloc/malloc.c: __malloc_initialized now signals three states:
	uninitialized, initializing, initialized.  Used in mcheck.

`__libc_malloc_initialized` is int, which is greater than 0 if initialized. So -1 means initializing as far as i looked at the code.

```suggestion
        return pwndbg.memory.s32(addr) > 0
```